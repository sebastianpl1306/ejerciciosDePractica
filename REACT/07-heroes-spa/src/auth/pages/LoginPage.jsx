import { useContext } from 'react';
import { useNavigate } from 'react-router-dom';
import { AuthContext } from '../context';

export const LoginPage = () => {
  const { login } = useContext( AuthContext );
  const navigate = useNavigate();

  const onLogin = () => {
    const lastPath = localStorage.getItem('lastPath');
    login('Sebastian Pabon Lopez');

    navigate(lastPath,{
      replace: true
    });
  }

  return (
    <>
      <h1>LoginPage</h1>
      <hr />
      <button 
        className="btn btn-primary"
        onClick={ onLogin }
      >Login</button>
    </>
  )
}