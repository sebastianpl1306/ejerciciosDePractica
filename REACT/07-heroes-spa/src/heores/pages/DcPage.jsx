import { HeroList } from '../components'

export const DcPage = () => {
  return (
    <div className="row">
      <h1>DcPage</h1>
      <hr />
      <HeroList publisher="DC Comics"/>
    </div>
  )
}