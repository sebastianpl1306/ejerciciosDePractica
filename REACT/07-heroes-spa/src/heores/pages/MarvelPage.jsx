import { HeroList } from '../components';

export const MarvelPage = () => {
  return (
    <div className="row">
      <h1>MarvelPage</h1>
      <hr />
      <HeroList publisher="Marvel Comics"/>
    </div>
  )
}