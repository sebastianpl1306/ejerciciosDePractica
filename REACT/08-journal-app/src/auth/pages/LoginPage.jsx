import { useMemo } from 'react';
import { Link as RouterLink } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';

import { startGoogleSignIn, startSignInWithEmailPassword } from '../../store';
import { useForm } from '../../hooks';

import { Google } from '@mui/icons-material';
import { Alert, Button, Grid, Link, TextField, Typography } from '@mui/material';
import { AuthLayout } from '../layout/AuthLayout';

const initialForm = {
  email: '',
  password: ''
};

export const LoginPage = () => {
  const { status, errorMessage } = useSelector(state => state.auth);
  const dispatch = useDispatch();

  const isLogin = useMemo(() => status == 'checking', [status]);

  const { email, password, onInputChange } = useForm(initialForm);

  const onSubmit = (event) =>{
    event.preventDefault();
    dispatch( startSignInWithEmailPassword(email, password) );
  }

  const onGoogleSignIn = () =>{
    dispatch( startGoogleSignIn() );
  }

  return (
    <AuthLayout title="Login">
      <form onSubmit={ onSubmit } aria-label="submit-form">
        <Grid container>
          <Grid item xs={ 12 } sx={{ mt: 2 }}>
            <TextField 
              label="Correo" 
              type="email" 
              placeholder="correo@google.com"
              fullWidth
              name="email"
              value={email}
              onChange={onInputChange}
            />
          </Grid>
          <Grid item xs={ 12 } sx={{ mt: 2 }}>
            <TextField 
              label="Contraseña" 
              type="password"
              inputProps={{
                'data-testid': 'password'
              }}
              placeholder="Contraseña"
              fullWidth
              name="password"
              value={password}
              onChange={onInputChange}
            />
          </Grid>
          <Grid container sx={{ mt: 1, display: !!errorMessage ? '' : 'none'}}>
            <Grid item xs={ 12 } sm={ 12 }>
              <Alert severity="error">{ errorMessage }</Alert>
            </Grid>
          </Grid>
          <Grid container spacing={ 2 } sx={{ mb: 2, mt: 1 }}>
            <Grid item xs={ 12 } sm={ 6 }>
              <Button type="submit" variant="contained" fullWidth disabled={isLogin}>
                Login
              </Button>
            </Grid>
            <Grid item xs={ 12 } sm={ 6 }>
              <Button variant="contained" fullWidth onClick={ onGoogleSignIn } disabled={isLogin} aria-label="btn-google">
                <Google/> <Typography sx={{ ml: 1 }}>Google</Typography>
              </Button>
            </Grid>
          </Grid>
          <Grid container direction="row" justifyContent="end">
            <Link component={ RouterLink } color="inherit" to="/auth/register">
            Crear Una cuenta
            </Link>
          </Grid>
        </Grid>
      </form>
    </AuthLayout>
  )
}