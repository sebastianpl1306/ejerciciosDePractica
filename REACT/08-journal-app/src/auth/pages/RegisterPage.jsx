import { useMemo, useState } from 'react';
import { Link as RouterLink } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';

import { Alert, Button, Grid, Link, TextField, Typography } from '@mui/material';
import { AuthLayout } from '../layout/AuthLayout';
import { useForm } from '../../hooks';
import { startCreatingUserWithEmailPassword } from '../../store';

const formData = {
  displayName: '',
  email: '',
  password: ''
}

const formValidations = {
  displayName: [(value)=>value.length >= 2,'El nombre debe tener mas de 1 caracter'],
  email: [(value)=>value.includes('@'),'El correo debe tener un @'],
  password: [(value)=>value.length >= 7,'La contrase単a debe tener mas de 1 caracter']
}

export const RegisterPage = () => {
  const dispatch = useDispatch();

  const { 
      email, emailValid,
      password, passwordValid,
      displayName, displayNameValid,
      formState,
      onInputChange,
      isFormValid
  } = useForm( formData, formValidations );

  const [formSubmitted, setFormSubmitted] = useState(false);
  const { status, errorMessage } = useSelector(state => state.auth);
  const isCheckingStatus = useMemo(() => errorMessage == 'checking', [status]);

  const onSumbit = ( event ) =>{
    event.preventDefault();
    setFormSubmitted(true);
    if (!isFormValid) return;
    dispatch(startCreatingUserWithEmailPassword(formState));
  }

  return (
    <AuthLayout title="Crear Cuenta">
      <form onSubmit={ onSumbit } >
        <Grid container>
          <Grid item xs={ 12 } sx={{ mt: 2 }}>
            <TextField 
              label="Nombre Completo" 
              type="text" 
              placeholder="Sebastian Pabon"
              fullWidth
              name="displayName"
              value={ displayName }
              onChange={ onInputChange }
              error={ !!displayNameValid && formSubmitted }
              helperText={ formSubmitted && displayNameValid }
            />
          </Grid>
          <Grid item xs={ 12 } sx={{ mt: 2 }}>
            <TextField 
              label="Correo" 
              type="text" 
              placeholder="ejemplo@dominio.com"
              fullWidth
              name="email"
              value={ email }
              onChange={ onInputChange }
              error={ !!emailValid && formSubmitted }
              helperText={ formSubmitted && emailValid }
            />
          </Grid>
          <Grid item xs={ 12 } sx={{ mt: 2 }}>
            <TextField 
              label="Contrase単a" 
              type="password" 
              placeholder="Contrase単a"
              fullWidth
              name="password"
              value={ password }
              onChange={ onInputChange }
              error={ !!passwordValid && formSubmitted }
              helperText={ formSubmitted && passwordValid }
            />
          </Grid>
          <Grid container spacing={ 2 } sx={{ mb: 2, mt: 1 }}>
            <Grid item xs={ 12 } sm={ 12 } sx={{display: !!errorMessage ? '' : 'none'}}>
              <Alert severity="error">{ errorMessage }</Alert>
            </Grid>
            <Grid item xs={ 12 } sm={ 12 }>
              <Button variant="contained" fullWidth type="submit" disabled={ isCheckingStatus }>
                Crear Cuenta
              </Button>
            </Grid>
          </Grid>
          <Grid container direction="row" justifyContent="end">
            <Typography sx={{ mr:1 }}>多Ya tienes cuenta?</Typography>
            <Link component={ RouterLink } color="inherit" to="/auth/login">
              Ingresar
            </Link>
          </Grid>
        </Grid>
      </form>
    </AuthLayout>
  )
}