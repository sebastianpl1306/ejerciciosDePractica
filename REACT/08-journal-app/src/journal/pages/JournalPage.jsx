import { IconButton } from '@mui/material';
import { JournalLayout } from '../layout/JournalLayout';
import { NoteView, NothingSelectedView } from '../views';
import { AddOutlined } from '@mui/icons-material';
import { startNewNote } from '../../store/journal';
import { useDispatch, useSelector } from 'react-redux';

export const JournalPage = () => {
  const { isSaving, active } = useSelector(state => state.journal);
  const dispatch = useDispatch();

  const onAddNewNote = () =>{
    dispatch(startNewNote());
  }

  return (
    <JournalLayout>
      {
        !!active ? <NoteView/> : <NothingSelectedView/>
      }
      <IconButton
        size='large'
        sx={{
          color: 'white',
          backgroundColor: 'error.main',
          ':hover': {backgroundColor: 'error.main', opacity: 0.9},
          position: 'fixed',
          right: 50,
          bottom: 50
        }}
        disabled={isSaving}
        onClick={ onAddNewNote }
      >
        <AddOutlined sx={{fontSize: 30}}/>
      </IconButton>
    </JournalLayout>
  )
}