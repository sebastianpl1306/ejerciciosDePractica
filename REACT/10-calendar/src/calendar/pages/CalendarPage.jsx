import { useState } from 'react';
import { Calendar } from 'react-big-calendar';
import 'react-big-calendar/lib/css/react-big-calendar.css';

import { CalendarEvent, CalendarModal, FabAddNew, FabDelete, Navbar } from '../components';
import { localizer, getMessagesES } from '../helpers';
import { useCalendarStore, useUiStore } from '../../hooks';

export const CalendarPage = () => {

  const { events, setActiveEvent } = useCalendarStore();
  const [lastView, setLastView] = useState(localStorage.getItem('lastView') || 'week');
  const { openDateModal } = useUiStore();

  const eventStyleGetter = ( event, start, end, isSelected ) =>{

    const style= {
      backgroundColor: '#347CF7',
      borderRadius: '0px',
      opacity: 0.8,
      color: 'white'
    }

    return{
      style
    }
  }
  
  const onDoubleClick = ( event ) =>{
    openDateModal();
  }

  const onSelect = ( event ) =>{
    setActiveEvent(event);
  }

  const onViewChanged = ( event ) =>{
    localStorage.setItem('lastView', event );
    setLastView( event );
  }

  return (
    <>
      <Navbar/>
      <Calendar
        localizer={localizer}
        culture='es'
        events={ events }
        startAccessor="start"
        defaultView={ lastView }
        endAccessor="end"
        style={{ height: 'calc( 100vh - 80px)' }}
        messages={ getMessagesES() }
        eventPropGetter={ eventStyleGetter }
        components={{
          event: CalendarEvent
        }}
        onDoubleClickEvent={ onDoubleClick }
        onSelectEvent={ onSelect }
        onView={ onViewChanged }
      />
      <CalendarModal/>
      <FabAddNew/>
      <FabDelete/>
    </>
  )
}