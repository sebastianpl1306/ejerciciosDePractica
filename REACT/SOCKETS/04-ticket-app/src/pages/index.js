export * from './ColaPage';
export * from './CrearTicketPage';
export * from './EscritorioPage';
export * from './IngresarPage';