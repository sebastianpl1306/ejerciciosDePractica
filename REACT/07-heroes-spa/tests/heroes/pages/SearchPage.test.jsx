import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { SearchPage } from '../../../src/heores/pages/SearchPage';

const mockUseNavigate = jest.fn();

jest.mock('react-router-dom',()=>({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockUseNavigate,
}));

describe('Pruebas en <SearchPage/>', () => {
    beforeEach(()=> jest.clearAllMocks());

    test('Debe de mostrarse correctamente con valores por defecto', () => {
        const { container } = render(
            <MemoryRouter>
                <SearchPage/>
            </MemoryRouter>
        )

        expect( container ).toMatchSnapshot();
    })

    test('Debe de mostrarse el hero cuando tenga el query parameter', () => {
        render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage/>
            </MemoryRouter>
        )

        expect( screen.getByText('Batman') ).toBeTruthy();

        const input = screen.getByRole('textbox');
        expect( input.value ).toBe('batman');

        const img = screen.getByRole('img');
        expect( img.src ).toContain('/heroes/dc-batman.jpg');

        const divSearch = screen.getByLabelText('div-serch-hero');
        expect( divSearch.style.display ).toContain('none');

        const divNoHero = screen.getByLabelText('div-no-hero');
        expect( divNoHero.style.display ).toContain('none');
    })

    test('Debe de mostrar un error si no encuentra el hero (batman123)', () => {
        render(
            <MemoryRouter initialEntries={['/search?q=batman123']}>
                <SearchPage/>
            </MemoryRouter>
        )

        const divSearch = screen.getByLabelText('div-serch-hero');
        expect( divSearch.style.display ).toContain('none');

        const divNoHero = screen.getByLabelText('div-no-hero');
        expect( divNoHero.style.display ).toContain('');
    })

    test('Debe de llamar el navigate a la pantalla nueva', () => {
        render(
            <MemoryRouter initialEntries={['/search']}>
                <SearchPage/>
            </MemoryRouter>
        )

        const input = screen.getByRole('textbox');
        fireEvent.change(input, {target: {value: 'batman'}});

        const form = screen.getByRole('form');
        fireEvent.submit(form);

        expect( mockUseNavigate ).toBeCalledWith('?q=batman');
    })
});