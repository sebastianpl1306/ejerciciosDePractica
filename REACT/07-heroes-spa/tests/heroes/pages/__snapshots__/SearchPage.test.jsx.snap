// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage/> Debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        action=""
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        aria-label="div-serch-hero"
        class="alert alert-primary"
      >
        Searh a hero
      </div>
      <div
        aria-label="div-no-hero"
        class="alert alert-danger"
        style="display: none;"
      >
        No hero with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
